### ISSUE TITLE
KubernetesExecutor. Pod log do not contains all data from task log file

### ISSUE THREAD

**Issue by KulykDmytro** [apache/airflow#11629]
**Apache Airflow version**: 1.10.12
**Kubernetes version**: v1.18.6
**What happened**:
Using KubernetesExecutor with **pod_template** to run tasks
pod log do not contains same info as shown in task log 

**What you expected to happen**:
Pod logs should contain all information from respective logs

**Anything else we need to know**:
Pod log available by api
```
C:\Users\dmytro.kulyk>kubectl logs aggregatesloaderdailycheckpaymentlegacy-a437af6584ed4006a0c62b07508f923e
[2020-10-18 10:36:25,449] {__init__.py:50} INFO - Using executor LocalExecutor
[2020-10-18 10:36:25,450] {dagbag.py:417} INFO - Filling up the DagBag from /opt/airflow/dags/aggregates_daily.py
```
Task log inside container
```
airflow@aggregatesloaderdailycheckpaymentlegacy-a437af6584ed4006a0c62b0:/opt/airflow$ cat logs/aggregates_loader_daily/check_payment_legacy/2020-09-13T02\:00\:00+00\:00/8.log 
[2020-10-18 10:36:25,834] {taskinstance.py:670} INFO - Dependencies all met for <TaskInstance: aggregates_loader_daily.check_payment_legacy 2020-09-13T02:00:00+00:00 [queued]>
[2020-10-18 10:36:25,867] {taskinstance.py:670} INFO - Dependencies all met for <TaskInstance: aggregates_loader_daily.check_payment_legacy 2020-09-13T02:00:00+00:00 [queued]>
[2020-10-18 10:36:25,868] {taskinstance.py:880} INFO - 
--------------------------------------------------------------------------------
[2020-10-18 10:36:25,868] {taskinstance.py:881} INFO - Starting attempt 8 of 9
[2020-10-18 10:36:25,868] {taskinstance.py:882} INFO - 
--------------------------------------------------------------------------------
[2020-10-18 10:36:25,890] {taskinstance.py:901} INFO - Executing <Task(SqlSensor): check_payment_legacy> on 2020-09-13T02:00:00+00:00
[2020-10-18 10:36:25,893] {standard_task_runner.py:54} INFO - Started process 15 to run task
[2020-10-18 10:36:25,915] {standard_task_runner.py:77} INFO - Running: ['airflow', 'run', 'aggregates_loader_daily', 'check_payment_legacy', '2020-09-13T02:00:00+00:00', '--job_id', '10678', '--pool', 'default_pool', '--raw', '-sd', 'DAGS_FOLDER/aggregates_daily.py', '--cfg_path', '/tmp/tmp6cm9p7nx']
[2020-10-18 10:36:25,916] {standard_task_runner.py:78} INFO - Job 10678: Subtask check_payment_legacy
[2020-10-18 10:36:25,992] {logging_mixin.py:112} INFO - Running %s on host %s <TaskInstance: aggregates_loader_daily.check_payment_legacy 2020-09-13T02:00:00+00:00 [running]> aggregatesloaderdailycheckpaymentlegacy-a437af6584ed4006a0c62b0
[2020-10-18 10:36:26,061] {base_hook.py:89} INFO - Using connection to: id: dwh. Host: dev.pgsql-dwh.betlab.private, Port: 5432, Schema: dwh, Login: airflow_user, Password: XXXXXXXX, extra: XXXXXXXX
[2020-10-18 10:36:26,062] {sql_sensor.py:89} INFO - Poking:  
                  select 1
                    from rawdata.bi_getpaymentssummary
                   where datepayment >= '2020-09-13'::timestamp
                     and datepayment < '2020-09-13'::timestamp + '1 day'::interval
                   limit 1      
     (with parameters None)
[2020-10-18 10:36:26,080] {base_hook.py:89} INFO - Using connection to: id: dwh. Host: dev.pgsql-dwh.betlab.private, Port: 5432, Schema: dwh, Login: airflow_user, Password: XXXXXXXX, extra: XXXXXXXX
....
```

**Comment by obrie** [comment]
I don't know that it's the appropriate solution, but for what it's worth this is the patch I've applied to my Airflow cluster to get both pod and task logs visible in running tasks:

```python
from airflow.configuration import conf
from airflow.utils.log.file_task_handler import FileTaskHandler
import os
original__read = FileTaskHandler._read


def _read(self, ti, try_number, metadata=None):
    log_relative_path = self._render_filename(ti, try_number)
    location = os.path.join(self.local_base, log_relative_path)

    if not os.path.exists(location) and conf.get('core', 'executor') == 'KubernetesExecutor':
        log = '*** Trying to get logs from worker pod {} ***\n\n'\
                .format(ti.hostname)

        try:
            from airflow.kubernetes.kube_client import get_kube_client
            from kubernetes.stream import stream as kubernetes_stream

            # Fix hostnames that exceeded max
            kube_client = get_kube_client()
            if len(ti.hostname) >= 63:
                pods = kube_client.list_namespaced_pod(conf.get('kubernetes', 'namespace'))
                matches = [
                    p.metadata.name for p in pods.items if p.metadata.name.startswith(ti.hostname)
                ]
                if len(matches) == 1 and len(matches[0]) > len(ti.hostname):
                    ti.hostname = matches[0]

            # Get Pod logs
            res = kube_client.read_namespaced_pod_log(
                name=ti.hostname,
                namespace=conf.get('kubernetes', 'namespace'),
                container='base',
                follow=False,
                tail_lines=100,
                _preload_content=False,
            )
            for line in res:
                log += line.decode()

            # Get Task logs
            resp = kubernetes_stream(
                kube_client.connect_get_namespaced_pod_exec,
                ti.hostname,
                conf.get('kubernetes', 'namespace'),
                command=['cat', location],
                stderr=True,
                stdin=False,
                stdout=True,
                tty=False,
                _preload_content=False,
            )
            try:
                while resp.is_open():
                    resp.update(timeout=1)
                    if resp.peek_stdout():
                        log += resp.read_stdout()
                    if resp.peek_stderr():
                        log += resp.read_stderr()
            finally:
                resp.close()
        except Exception as f:
            log += '*** Unable to fetch logs from worker pod {} ***\n{}\n\n'.format(
                ti.hostname, str(f)
            )

        return log, {'end_of_log': True}
    else:
        return original__read(self, ti, try_number, metadata)


FileTaskHandler._read = _read
```

**Comment by viniciusdsmello** [comment]
Hi Guys, any updates here? I was wondering if it is possible to read tasks' logs through UI during their execution. When a task is running, I can't see the logs by examining the Pod logs neither at UI, It's available for me only after the tasks finishes.

I'm asking this because sometimes I need to check the status/logs of a given task that may take a few hours to complete.

**Comment by KulykDmytro** [comment]
logs are currently available under task log within pods filesystem but not the pod's log 
```
logs/{dag_id}/{task_id}/{task_run_id}.log
```

**Comment by potiuk** [comment]
it appears this issue was resolved in PR https://github.com/apache/airflow/pull/17649

that PR was released in airflow 2.0.3
